cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -61,36 +61,101 @@ const (
 	PoolPrefix PoolNamePrefix = "cstor-"
 )
 
+type ImportOptions struct {
+	CachefileFlag bool
+	DevPath string
+	dontImport bool
+}
+
 // RunnerVar the runner variable for executing binaries.
 var RunnerVar util.Runner
 
 // ImportPool imports cStor pool if already present.
-func ImportPool(cStorPool *apis.CStorPool, cachefileFlag bool) error {
-	importAttr := importPoolBuilder(cStorPool, cachefileFlag)
+func ImportPool(cStorPool *apis.CStorPool, importOptions *ImportOptions) (string, error) {
+	var importAttr []string
+
+	importAttr = importPoolBuilder(cStorPool, importOptions)
+
 	stdoutStderr, err := RunnerVar.RunCombinedOutput(zpool.PoolOperator, importAttr...)
 	if err != nil {
-		glog.Errorf("Unable to import pool: %v, %v", err.Error(), string(stdoutStderr))
-		return err
+		glog.Errorf("Unable to import pool with devPath: %v, %v devpath: %v cachefile: %v", err.Error(),
+			string(stdoutStderr), importOptions.DevPath, importOptions.CachefileFlag)
+		return string(stdoutStderr), err
 	}
-	glog.Info("Importing Pool Successful")
-	return nil
+
+	glog.Infof("Import command successful with %v %v dontimport: %v", importOptions.CachefileFlag, importOptions.DevPath, importOptions.dontImport)
+	return string(stdoutStderr), nil
 }
 
 // importPoolBuilder is to build pool import command.
-func importPoolBuilder(cStorPool *apis.CStorPool, cachefileFlag bool) []string {
+func importPoolBuilder(cStorPool *apis.CStorPool, importOptions *ImportOptions) []string {
 	// populate pool import attributes.
 	var importAttr []string
+	var cachefileFlag bool
+	var devPath string
+
+	// import takes either cachefile or devPath, not both.
+	// here, prioritizing devPath. Using zero values for cachefileFlag and devPath
+	if importOptions.DevPath != "" {
+		devPath = importOptions.DevPath
+	} else {
+		cachefileFlag = importOptions.CachefileFlag
+	}
+
 	importAttr = append(importAttr, "import")
 	if cStorPool.Spec.PoolSpec.CacheFile != "" && cachefileFlag {
 		importAttr = append(importAttr, "-c", cStorPool.Spec.PoolSpec.CacheFile,
 			"-o", cStorPool.Spec.PoolSpec.CacheFile)
 	}
-	importAttr = append(importAttr, string(PoolPrefix)+string(cStorPool.ObjectMeta.UID))
+	if devPath != "" {
+		importAttr = append(importAttr, "-d", devPath)
+	}
+
+	if importOptions.dontImport == false {
+		importAttr = append(importAttr, string(PoolPrefix)+string(cStorPool.ObjectMeta.UID))
+	}
 	return importAttr
 }
 
+// GetDevPath gets the path from given deviceID
+func GetDevPath(devid string) string {
```

Comment:
Can devPath be obtained from devID ( which is devlink) by manipulating strings ? 
Also -- will this be consistent across platforms ?

Reply:
its under assumption that NDM gives us absolute path of the disk